x-pack/modules/azure/lib/filters/azure_event.rb

```diff
@@ -0,0 +1,246 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+require "logstash/filters/base"
+require "logstash/namespace"
+require "logstash/json"
+
+module LogStash
+  module Filters
+    class AzureEvent < LogStash::Filters::Base
+
+      config_name "azure_event"
+
+      # When all heuristics fail to identify a category, this category will be used if and only if the default_match fields are present, and the resource has been parsed correctly
+      config :default_category, :validate => :string, :default => "Administrative"
+
+      # These fields must be present to match to the default default category
+      config :default_match, :validate => :array, :default => ["operationName", "level"]
+
+      # Possible [azure][category] values for [azure][group] = "activity_log", the category names must be mutually exclusive between groups
+      config :activity_log_categories, :validate => :array, :default => ["Administrative", "ServiceHealth", "Alert", "Autoscale", "Security"]
+
+      # Possible [azure][category] values for [azure][group] = "sql_diagnostics", the category names must be mutually exclusive between groups
+      config :sql_diagnostics_categories, :validate => :array, :default =>
+          ["Metric", "Blocks", "Errors", "Timeouts", "QueryStoreRuntimeStatistics", "QueryStoreWaitStatistics", "DatabaseWaitStatistics", "SQLInsights"]
+
+      # Append values to the `tags` field when there has been not been a match to a known azure event group or category.
+      config :tag_on_failure, :validate => :array, :default => ["_azure_event_failure"]
+
+      ACTIVITY_LOG_GROUP = "activity_log".freeze
+      SQL_DIAGNOSTICS_GROUP = "sql_diagnostics".freeze
+
+      def register
+        # do nothing
+      end
+
+      def filter(event)
+        resource_id = event.get("resourceId")
+        if resource_id && add_resources(resource_id, event) && add_group_and_category(event)
+          # rename properties to include the group/category to help avoid mapping errors
+          if event.get("properties")
```

Comment:
in what situation would this be false? I see the event is seen as valid (`filter_matched` is applied)

Reply:
Azure events are valid with or without a `properties` field.  About 1/2 of the events have it, the rest don't.